### PR TITLE
added msr trap support

### PR DESCRIPTION
**Pull request by archercreat**
Hello! I was surprised there were no MSR traps, so I added them :)

### CODE REVIEW

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Comment by archercreat** [comment]
@tklengyel Fixed style issue.

**Comment by tklengyel** [comment]
@drakvuf-jenkins Retest this please

File: src/libdrakvuf/libdrakvuf.h [hunk 1]
```diff
@@ -201,6 +201,7 @@ typedef struct drakvuf_trap_info
     {
         const cpuid_event_t* cpuid; /* For CPUID traps */
         const debug_event_t* debug; /* For DEBUG traps */
+        const reg_event_t*   reg;   /* For MSR traps */
```
Comment (tklengyel) [inline src/libdrakvuf/libdrakvuf.h:204]: I think calling this field `msr` here would be more intuitive.

Comment (archercreat) [reply src/libdrakvuf/libdrakvuf.h:204]: Right now a user would access msr in a callback like this:
```c++
auto msr = info->reg->msr;
```
IMO this variant looks better, but I'm ok with the change.

Comment (tklengyel) [reply src/libdrakvuf/libdrakvuf.h:204]: Yea, I guess it's fine as-is.

**Comment by tklengyel** [comment]
Thanks!

**Pull request merged**